config.yml

```diff
@@ -13,6 +13,8 @@ defaults:
     allowedPatterns:
       - '^https://www.mozilla.org/'
       - '^https://s3-us-west-2.amazonaws.com/taskcluster-public-artifacts/'
+      - '^https://public-artifacts.taskcluster.net/'
```

Comment:
Yep.  I was a little weary of using Urls in the first place.  I would like to have it possible to say allow `http://www.example.org:80/a/` but block `http://www.example.org:80/b/`

I think the main danger here is that someone could write a bad regex that is like this `^https://www.google.com`, but then an attacker could request `https://www.google.com.my.evil.domain/my-blog-to-host-for-freeeeeee`.

What about having a simple pattern validator that enforces that all patterns start with `^` and end with `/`.  I think I already normalize the URL before checking against the pattern, so we should be good for that.